lib/min_heap.rb

```diff
@@ -8,24 +8,52 @@ def initialize(key, value)
 end
 
 class MinHeap
+  attr_reader :store
 
   def initialize
     @store = []
   end
 
   # This method adds a HeapNode instance to the heap
-  # Time Complexity: ?
-  # Space Complexity: ?
+  # Time Complexity: usually O(1) because unlike adding to the front, appending /
+  # adding to the end doesn't require reordering the whole array
+  # Space Complexity: O(1)
   def add(key, value = key)
```

Comment:
The time complexity is O(log n) because the heap_up method is O(log n).  Because it's recursive, the space complexity is also O(log n)

Otherwise this works.